posts/2025-06-03-25.0.0.6-beta.adoc

```diff
@@ -0,0 +1,187 @@
+
+This beta now introduces a new `server.xml` attribute for the `<mpHealth>` element named `startupCheckInterval` with an associated `MP_HEALTH_STARTUP_CHECK_INTERVAL` environment variable. This configuration is optional and defaults to 100 ms if the user has not configured it. The configuration accepts a number followed by an optional `ms` or `s` to denote the time unit. If no time unit is specified, it defaults to milliseconds. Invalid values default the interval to 100 ms.  
+
+The `startupCheckInterval` is used during the _startup_ phase of the file-based health check mechanism. As a refresher, the file-based health check functionality creates the `live`, `ready`, and `started` files under the `${server.output.dir}/health` directory. The `${server.output.dir}` defaults to `wlp/usr/servers/<server>`. For more information, see link:{url-prefix}/docs/latest/reference/directory-locations-properties.html[directory locations and properties]. 
+
+These files are created when their respective health checks return an `UP` status. The `startupCheckInterval` is the interval at which the runtime queries the health checks during the startup phase. When all statuses report `UP` then all three `started`, `ready` and `live` files are created. Next, proceed to the _update_ phase. At this point, `started` file is no longer useful. After Kubernetes has found an `UP` status, it proceeds to start its checks for _liveness_ and _readiness_. It's at this point when the `checkInterval` is used by the runtime to establish the interval at which the liveness and readiness health checks are queried. Similar to before, if an `UP` status is resolved then the respective health-check file is _updated_ (that is, the last modified time of the file is updated). A `DOWN` status results in the respective health-check file not being updated.
+
+The following example shows the `server.xml` configuration with both configurations:
+Note that only `checkInterval` is required to enable the functionality and the `startupCheckInterval` is optional as it defaults to `100 ms`.
+The `startupCheckInterval` is faster compared to the `checkInterval` to allow for the creation of the files as soon as possible.
+
+[source,xml]
+```
+<server>
+  <features>
+    <feature>mphealth-4.0</feature>
+  </features>
+  <mpHealth checkInterval="10s" startupCheckInterval="150ms"/>
+</server>
+```
+
+With file-based checks, `exec` command strategy can be used instead of the `httpGet` or `gRPC` strategy when configuring the Kubernetes health check probes. In this beta, scripts are introduced in to the Open Liberty(beta) image for these file-based health checks.
```

Comment:
```suggestion
With file-based checks, the `exec` command strategy can be used instead of the `httpGet` or `gRPC` strategy when configuring the Kubernetes health check probes. In this beta, scripts are introduced into the Open Liberty image for these file-based health checks.
```